### ISSUE TITLE
[Suggestion] Draconic Reactor has gotten power-creep'd

### ISSUE THREAD

**Issue by Buggy123** [brandon3055/Draconic-Evolution#950]
Personally, I love the Draconic Reactor. It's a interesting and complex take on end-game power production. It requires a great deal of materials and work to build, and potentially control circuitry of some sort to manage it. And the interactions between efficiency, temperature, etc add subtlety without forming a massive barrier to entry; easy to learn, hard to master. However, I can't help but notice that it's not actually... _powerful_.

Lots of packs include things like Big/Extreme Reactors, Nuclearcraft, Mekanism, and more. Extreme reactors can get you [2 million or more](http://br.sidoh.org/#reactor-design?length=30&width=30&height=46&activelyCooled=false&controlRodInsertion=7&layout=C4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC7XC4XC4XC4XC4XC4XC2XC4XC4XC4XC4XC4XC2X) at max size, and if you go for active cooling+a Mekanism turbine you can make one much more space and fuel efficient. Mekanism fusion reactors can produce _[fifty million RF/T](https://imgur.com/a/Nsag5)_ with a moderate setup, and I've personally created setups which produce over _100 million RF/T_. I mean, the latest version of Environmental Tech has a _solar panel_ that produces [1.7 million RF/T](https://imgur.com/zVdU1eV.jpg)!

All of these are safe, relatively easy to set up (compared to a Draconic Reactor, at least), require little upkeep, and produce more than most Draconic Reactors can unless you're very careful or very crazy. On the other hand, Draconic Reactors require quite a bit of resources to setup, some of which can be nigh-impossible to automate, requires some level of control circuitry, and will violently destroy your entire base if anything goes wrong. The only place where they come ahead is visuals and fuel efficiency.

I _like_ Draconic Reactors, but I'm seeing a lot of cons and not a lot of pros. I don't see why a person would use one when _any_ of these common alternatives are available, unless they like the awesome visuals enough to risk their entire base exploding. And this is all ignoring the fact that Draconic Evolution is the go-to OP mod. Armor, weapons, tools, power networks, power storage... everything is really hard to get and really powerful, and it stands out above all other mods in that regard... except for power generation.

In summary, the Draconic Reactor needs a rebalance.

I'm not sure what form this would take. I _don't_ think just multiplying the power output of the Reactor is the right solution. I was thinking that it'd be best if the mechanics were tweaked slightly, so that there isn't a soft upper limit to power generation; you can generate as much power as you want, so long as you manage it properly and have enough fuel. Furthermore, you could allow it to refuel while running, so that you are rewarded with higher fuel efficiency for keeping it running. But you make the mod after all, it's all up to you.

**Comment by ruifung** [comment]
While all of that sounds easy, from my experience poking around the reactor calculation code when writing my OC program for reactor control, this small tweak would probably mean changing how output power is calculated entirely here.

That being said, it's true that it will destroy your entire base if anything goes wrong, but it's *possible* to defend against the explosion to some extent. Although, it will likely leave your base surrounded by a crater.

**Comment by Buggy123** [comment]
True, rebalancing it might be difficult, but at this point its pretty much that or let it fall into obsolescence. 

Playing ATM3, for similar effort to making a draconic reactor I can build a single solar panel multiblock that will never, ever explode or run out of fuel, and put out 1.7m RF/T during the day. Or a modest Mekanism fusion reactor, and make at least that much. Or a extreme reactor. Or a nuclearcraft reactor. None of these require a control mechanism, and I don't have to spend any time or resources protecting my base from them.

I just don't see why anyone is going to use a DE reactor if any of those alternatives are present (and they usually are). Choosing form over function is _not_ a terribly common choice. With mods like Immersive Engineering, some people make packs where everything is toned down so that it is actually useful. But this is **Draconic Evolution**. No one includes DE in a pack if they want a modest endgame.

**Comment by VADemon** [comment]
When Applied Energistics first released it was the most OP thing I've ever seen. One system - connect to as many inventories as you wish, no hassle. Some time later, with AE2 they "fixed" or rebalanced their whole mod and made the convenience not so easy to achieve.

Moral of the story: Is it really DE's fault that mods like BigReactors output so much energy in their default mod-pack shipped configuration?

**Comment by Buggy123** [comment]
No, but fault isn't really a issue.

Other mods don't necessarily have this issue of obsoletion, because of modpacks designing around them so that they aren't obsoleted. For instance, Immersive Engineering diesel generators may not be great compared to other generators, but some modpacks balance around Immersive Engineering and avoid or nerf more overpowered mods, such that it's still useful.

However, this won't happen for DE because everything _else_ about the mod is so strong; draconic armor remains largely unbested over the years, the grinder is the fastest grinder out of pretty much all mods, draconic tools are the best tools, item attractors are the best magnets, and so on. No one will balance a modpack around DE because everything _except_ DE reactors is extremely powerful end-game stuff.

Quite simply, we have 2 choices here. We can leave reactors unchanged, and they will remain obsolete and almost unused in lieu of other, superior generation options. At best a novelty. Or, alternatively, we can revamp them to some degree, either directly buffing them or altering their mechanics so that they are able to compete again.

**Comment by brandon3055** [comment]
Ok so time to add my thoughts on this issue. 
The reactor is OP theirs no two ways about it. it "can" produce up to 2 million RF/t+ constantly and stable at the upper end of its efficiency curve. That is OP and that is the reason the reactor is so expensive and dangerous. 

The RF API was never meant to work with energy values this big which is one of the reasons i will be switching to a custom energy system in the next major DE re write. I do not want to further contribute to the inflation of RF which is why the reactor at its lowest efficiency only produces around 400,000 RF't. and the only reason it goes up to 2 million RF/t+ is i miscalculated the efficiency curve. I only intended it to go up to around  800,000 RF't

Mods with simple safe power gen that is capable of millions of RT per tick are just absolutely ridiculous. Solar panels are especially stupid and defy all logic. To produce 1million+ RF/t from solar you should have to cover half your f***ing world in solar panals. Craming all that into a single block is stupid and unrealistic to the point that the "its a game it does not need to emulate reality" argument no longer applies.

**Comment by ruifung** [comment]
By upper end of it's efficiency curve you mean, as fuel conversion rate increases?

To be honest, I quite enjoy having big, powerful, reactors that have devastating consequences,.
It doesn't have to be like completely uncontainable, but you just need to make sure you build layers upon layers of safeguards on it.

On another (unrelated) note, is it possible to initiate a shutdown of the reactor with a redstone signal? =D

**Comment by brandon3055** [comment]
Kind of. If you have failsafe mode engaged then you can use a redstone signal to shut down the output and as long as you have enough energy backlog to maintain the shield the reactor will automatically shut down once its saturation maxes out. Thats the purpose of failsafe mode.

**Comment by ruifung** [comment]
The problem with that is that in the event the output storage is full, the reactor needs to go through a full startup sequence again. I guess I'll just rig up 2 computers in parallel as a failsafe for the control program.
